### ISSUE TITLE
local time is preferred to be shown when do the docker inspect container operation

### ISSUE THREAD

**Issue by xingbao-hzj** [moby/moby#29236]
When I do the operation:docker inspect container-xxx,
the info shown  for field:"Created", "StartedAt","FinishedAt" is not in local time,but in UTC time.And it's not convenient for docker user.

Therefore I suggest change the time to be shown in local time.
What do you think of it.

Thanks.

**Comment by cpuguy83** [comment]
`docker inspect` is a low-level tool which sends back raw json.
I don't think we should change the details of the fields at the client.

**Comment by thaJeztah** [comment]
No there's no such flag, and for `docker inspect` I don't think we should add one, as it's quite out of scope for what that function is designed to do. Also see https://github.com/moby/moby/pull/35051#issuecomment-334554828 for a similar discussion for viewing logs.

I'll close this issue, because of the above, but feel free to comment after I closed

**Comment by mostolog** [comment]
Thanks @thaJeztah, for sure https://github.com/paultag/notes.pault.ag/blob/master/notes/its-all-relative.md was an interesting reading.

Although I would love to have a --print-timezoned option for _docker inspect_, and even considering the above article states:
```It's not until you want to show the datetime to the user again should you consider how to re-encode your datetime to bytes. You should think about what flavor of bytes, what encoding -- what timezone -- should I be encoding into?
```

I understand working in UTC would make more sense on a "globally managed docker environment" (with devops on different timezones), which sadly is not our use case, as we work all on the same country.


On the other hand, **having UTC timestamps in some places, while GMT+1 on another (server logs) is a prone-to-human-error scenario we would like to avoid.**

Would adding timezoned_fields to that JSON would be considered ok?
Perhaps an option to display those using server timezone would be better?
Maybe running dockerd with "--use-server-timezone" would solve those issues?

Awaiting your reply,
Regards

**Comment by thaJeztah** [comment]
@mostolog difference here is that the `inspect` output is meant to show what's stored, and used internally by the engine; having timestamps in different timezones is not something that's used by the engine, thus likely won't make much sense to add for presentation.

**Comment by mostolog** [comment]
As you provided enough feedback to agree dockerd shouldn't use timezones/locale but UTC, and considering using UTC on dockerd but locale/timezones for system logs (and I'll dare to say we agree that's a "prone-to-human-error" scenario), what do you think it could be the solution?

Developing our own inpect tool which converts to TZ?
Perhaps adding a localized version of that strings won't do so much harm...

Thanks

**Comment by orti99** [comment]
It is very frustrating that the docker log still doesn't show the correct (local) time. 
For the person inspecting log files it just doesn't make any sense that all host log files show different times than the docker logs and it causes confusing when you're not leaving in UTC.

Timezones should be valid for the whole system including runnings VMs.
